Add features to module file system

Pouzaudr opened the pull request:
This patch applies the changes:
- FILE_SEPARATOR in Kernel File System
- Remove directory in Kernel File System
- Correct the method join in paths.js to use FILE_SEPARATOR
- Correct the method requestResponse in dahuapp to use FILE_SEPARATOR

benabbon committed:
introduce a utility function in path :

getDahuFileFromDirectory : Get the name of the presentation file to
create when we create a new project, it is calculated from the name of the Directory.

benabbon committed:
introduce a Directory chooser window in the kernel

benabbon committed:
add the new project button in the menu

benabbon committed:
Handle the creation of a new project

benabbon committed:
Fortify conditions on screen view :

This is important for when we want to create a new project
with no screens on screencastn or when we initialize the workspace with
no screen.

barraq committed:
Merge branch 'next' into next-new-project

Conflicts:
	dahu/core/app/scripts/dahuapp.js
	dahu/editor/src/main/java/io/dahuapp/editor/DahuApp.java

barraq commented:
@Poupouz I merged the next-add-features-module-file-system with the latest version of Next (which contain the new Screencast controller). Please could you fix take into account my remarks so that I can do the merge between next and this branch.

benabbon committed:
migrate utility functions to screencast controller

the function moved here is getDahuFileFromDirectory, used
to get the name of the presentation to create relating to
the given directory

benabbon committed:
factorize layout creation code

this code is used when we open an existing project
or when we create a new empty one.

barraq committed:
Merge branch 'next-new-project' into next

Commit:
Merge branch 'next-add-features-module-file-system' of ssh://github.com/dahuapp/Dahu into next-add-features-module-file-system

Conflicts:
	dahu/core/app/scripts/modules/utils/paths.js

Commit:
Move controler

From reqResponse to screencast controller

barraq committed:
add Paths.dirname method to Paths helper
